trade_remedies_caseworker/organisations/views.py

```diff
@@ -261,6 +272,251 @@ def post(
+
+    # for use in form
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs["org_invite_contacts"] = self.get_org_invite_contacts()
+        return kwargs
+
+    def form_valid(self, form):
+        # selected contacts
+        self.request.session["selected_contacts"] = self.request.POST.getlist("which_contact")
+        return super().form_valid(form)
+
+    def get_next_url(self, form=None):
+        return reverse(
+            "organisations:invite-party-check",
+            kwargs={
+                "case_id": self.kwargs["case_id"],
+                "organisation_id": self.kwargs["organisation_id"],
+            },
+        )
+
+
+class OrganisationInviteContactNewView(BaseOrganisationInviteView):
+    template_name = "organisations/invite_party_contact_new.html"
+    form_class = OrganisationInviteContactNewForm
+
+    # for use in html template
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context.update(self.request.GET)
+
+        # self.request.session["new_contact"] might not be set yet
+        if self.request.session.get("new_contact") and "selected_contacts" in self.request.session:
+            contact = self.client.contacts(self.request.session["selected_contacts"][0])
+            context["selected_contact_name"] = contact.name
+            context["selected_contact_email"] = contact.email
+            context["selected_organisation"] = self.client.organisations(contact.organisation).name
+
+        return context
+
+    def form_valid(self, form):
+        # Create new organisation
+        organisation = self.client.organisations(
+            {"name": form.cleaned_data["organisation_name"]}, fields=["id"]
+        )
+        self.kwargs["organisation_id"] = organisation["id"]
+
+        # Create new contact and associate with (new) organisation
+        contact = self.client.contacts(
+            {
+                "name": form.cleaned_data["contact_name"],
+                "email": form.cleaned_data["contact_email"],
+                "organisation": organisation["id"],
+            }
+        )
+        self.request.session["selected_contacts"] = [self.client.contacts(contact["id"]).id]
+        self.request.session["new_contact"] = True
+        return super().form_valid(form)
+
+    def get_next_url(self, form=None):
+        return reverse(
+            "organisations:invite-party-check",
+            kwargs={
+                "case_id": self.kwargs["case_id"],
+                "organisation_id": self.kwargs["organisation_id"],
+            },
+        )
+
+
+class OrganisationInviteReviewView(BaseOrganisationInviteView):
+    template_name = "organisations/invite_party_check.html"
+    form_class = ValidationForm
+
+    def get_selected_contacts(self):
+        selected_contacts = []
+        if "selected_contacts" in self.request.session:
+            for contact_id in self.request.session["selected_contacts"]:
```

Comment:
instead of checking if "selected_contacts" is in self.request.session, we can use the .get() call to return the right value if it exists, or an empty list if it doesn't (meaning that it won't run the for loop). 

```suggestion
            for contact_id in self.request.session.get("selected_contacts", []):
```